app/controllers/services_controller.rb

```diff
@@ -1,147 +1,20 @@
 # :nodoc:
-class ServicesController < ManageServicesController
+class ServicesController < BasePublishedRecordController
   include ManageMetadataHelper
 
   before_action :set_service, only: [:show, :edit, :clone, :destroy, :revisions, :revert, :download_json]
   before_action :set_schema, only: [:show, :edit, :clone, :destroy]
-  before_action :ensure_correct_service_provider, only: [:edit, :clone, :destroy]
+  before_action :ensure_correct_provider, only: [:edit, :clone, :destroy]
   before_action :set_preview, only: [:show]
 
-  add_breadcrumb 'Services' # there is no services index action, so not providing a link
-
-  def show
-    if params[:not_authorized_request_params]
-      @record_action = 'manage-collection-associations' if params.fetch(:not_authorized_request_params, {})['controller'] == 'collection_associations'
-
-      set_user_permissions
-    end
-
-    @draft = ServiceDraft.where(provider_id: @provider_id, native_id: @native_id).first
-
-    add_breadcrumb breadcrumb_name(@service, 'service'), service_path(params[:id])
-  end
-
-  def edit
-    if @native_id
-      draft = ServiceDraft.create_from_service(@service, current_user, @native_id)
-      Rails.logger.info("Audit Log: Service Draft for #{draft.entry_title} was created by #{current_user.urs_uid} in provider #{current_user.provider_id}")
-      redirect_to service_draft_path(draft), flash: { success: I18n.t('controllers.draft.service_drafts.create.flash.success') }
-    else
-      Rails.logger.info("User #{current_user.urs_uid} attempted to edit Service #{@concept_id} in provider #{current_user.provider_id} but a Service Draft was not created to edit because there was no native_id (#{@native_id}) found.")
-      # if we cannot locate the native_id for the Service, we should discontinue editing
-      redirect_to service_path(@concept_id, revision_id: @revision_id), flash: { error: I18n.t('controllers.services.edit.flash.native_id_error') }
-    end
-  end
-
+  # https://stackoverflow.com/questions/34735540/action-defined-in-applicationcontroller-can-not-be-found
```

Comment:
So this link actually does not seem to quite explain the issue. I would rather you just write a comment describing the error raised, and that we have looked into it and couldn't get it to work without the declaring the method.